caffe2/python/onnx/frontend.py

```diff
@@ -211,7 +212,37 @@ def apply_trans(k, dim=2, ks=None):
         apply_trans('stride')
         apply_trans('dilation')
         apply_trans('adj')
-        apply_trans('pad', 4)
+        apply_trans('pad', dim=4)
+
+        legacy_pad_attr = attrs.pop('legacy_pad', None)
+        if legacy_pad_attr:
+            # The problem here is that, Pool op in Caffe may add an additional pixel, if the last part is smaller than stride.
+            # So we use the explicit padding to replace legacy_pad.
+            # pad[end] = output_size[start + 2] * stride[start] - pad[start] - 1 + kernel[start] - input[start + 2]
+            # end = start + len(pad) / 2
+            assert node.op_type.endswith("Pool")
+            assert not node.op_type.startswith("Global")
+            input_size = shapes[node.input[0]]
+            output_size = shapes[node.output[0]]
+            assert len(output_size) == 4
+            if legacy_pad_attr.i == 1:
+                attrs.pop('pads')
```

Comment:
`pads` and `legacy_pad` should not be set simultaneously:
https://github.com/caffe2/caffe2/blob/02ac8365935bcaf5f6552ba6d87f1285631a219b/caffe2/operators/conv_pool_op_base.h#L105